src/sas/qtgui/Plotting/UnitTesting/PlotterTest.py

```diff
@@ -444,6 +444,48 @@ def notestOnModifyPlot(self):
             self.plotter.onModifyPlot(2)
         self.plotter.figure.clf()
 
+    def testOnToggleLegend(self):
+        """
+        Make sure Legend can be switched on/off
+        """
+        # Prepare new data
+        data2 = Data1D(x=[1.0, 2.0, 3.0],
+                       y=[11.0, 12.0, 13.0],
+                       dx=[0.1, 0.2, 0.3],
+                       dy=[0.1, 0.2, 0.3])
+        data2.title="Test data 2"
+        data2.name="Test name 2"
+        data2.id = 2
+        data2.custom_color = None
+        data2.symbol = 1
+        data2.markersize = 11
+
+        self.plotter.plot(data2)
+
+        self.assertTrue(self.plotter.showLegend)
+        # assure we see the legend
+        self.assertTrue(self.plotter.legend.get_visible())
+
+        # toggle legend
+        self.plotter.onToggleLegend()
+
+        # now we don't see the legend
+        self.assertFalse(self.plotter.legend.get_visible())
+
+        # toggle again
+        self.plotter.onToggleLegend()
+
+        # see the legend again
+        self.assertTrue(self.plotter.legend.get_visible())
+
+        # switch the visibility of the legend
+        self.plotter.showLegend = False
```

Comment:
Have you tried turning it off and turning it back on again? In all seriousness, this is a good, simple test.